python/graphscope/client/session.py

```diff
@@ -913,20 +918,36 @@ def gremlin(self, graph, engine_params=None):
             graph.vineyard_id in self._interactive_instance_dict
             and self._interactive_instance_dict[graph.vineyard_id] is not None
         ):
-            return self._interactive_instance_dict[graph.vineyard_id]
+            interactive_query = self._interactive_instance_dict[graph.vineyard_id]
+            if interactive_query.status == "running":
+                return interactive_query
+            elif interactive_query.status == "pending":
+                begin_time = time.time()
+                while True:
+                    time.sleep(1)
+                    if interactive_query.status == "running":
+                        return interactive_query
+                    if time.time() - begin_time >= timeout_seconds:
+                        raise TimeoutError("Launch gremlin server timeout")
 
         if not graph.loaded():
             raise InvalidArgumentError("The graph has already been unloaded")
         if not graph.graph_type == types_pb2.ARROW_PROPERTY:
             raise InvalidArgumentError("The graph should be a property graph.")
 
+        from graphscope.interactive.query import InteractiveQuery
+
+        interactive_query = InteractiveQuery(
+            graphscope_session=self, object_id=graph.vineyard_id
+        )
+        self._interactive_instance_dict[graph.vineyard_id] = interactive_query
```

Comment:
If the interactive_instance creating eventually failed, should the item in dict be removed?

Reply:
Will set the status to `Failed` if create failed, So we rethrow exception when users call .gremlin() again.